### DISCUSSION TITLE
How to create .a library of c++ headers for wasm?

### DISCUSSION THREAD

**Discussion by riajain0412** [unicode-org/icu4x#3857] [Q&A]
For using ICU4X in c++, we can build .a library with the following command `cargo build -p icu_capi_staticlib --all-features`. I want to build ICU4X for wasm so how can I create this static library for wasm?

**Comment by sffc** [comment]
Build `icu_capi_cdylib` with `--target wasm32-unknown-unknown`

For examples, see https://github.com/unicode-org/icu4x/tree/main/ffi/diplomat/js/examples

**Reply by riajain0412** [reply]
With this command .a files is not getting created.

**Reply by Manishearth** [reply]
Please be more specific about what you are trying to do, if you're linking to c++ you should look at the other example I posted

**Reply by riajain0412** [reply]
I'm trying to build ICU4X for wasm where I want to link .a(static library) to my sample. I was able to create .a library in case of macos but it is not getting created for wasm.

**Reply by Manishearth** [reply]
.a libraries do not contain headers.

Are you trying to call ICU4X from JS, or from C++? If you build with `wasm32-unknown-emscripten` you will get a `.a` file you can link to C++.

**Reply by riajain0412** [reply]
I'm trying to call it from C++. 
 .a file got created but it does not contain following symbol `_ICU4XDataProvider_create_test`

**Comment by Manishearth** [comment]
If you want to use it directly from JS, @sffc's answer is correct.

If you want to use it from C++ but your C++ application is being built to wasm, use the emscripten target instead. See https://github.com/unicode-org/icu4x/tree/main/ffi/diplomat/cpp/examples/fixeddecimal_wasm for an example.

**Comment by riajain0412** [comment]
> .a libraries do not contain headers.
> 
> Are you trying to call ICU4X from JS, or from C++? If you build with `wasm32-unknown-emscripten` you will get a `.a` file you can link to C++.

Yeah right, I wrote it by mistake.
